File: LearningLens2025/legacy_files/focused-ai-ui/lib/models/course.dart
class Course {
  final String id;
  final String name;
  final String description;
  final String platform;
  final String? instructor;
  final int? enrollmentCount;
  final DateTime? createdAt;

  Course({
    required this.id,
    required this.name,
    required this.description,
    required this.platform,
    this.instructor,
    this.enrollmentCount,
    this.createdAt,
  });

  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      platform: json['platform'] ?? '',
      instructor: json['instructor'],
      enrollmentCount: json['enrollmentCount'],
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'platform': platform,
      'instructor': instructor,
      'enrollmentCount': enrollmentCount,
      'createdAt': createdAt?.toIso8601String(),
    };
  }
}

File: LearningLens2025/legacy_files/focused-ai-caila-service/caila-service/caila-service/src/main/java/com/focusedai/caila/models/domain/Course.java
package com.focusedai.caila.models.domain;

import lombok.Data;
import lombok.Builder;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Course {
    private String id;
    private String name;
    private String description;
    private String platform;
    private String instructor;
    private Integer enrollmentCount;
    private LocalDateTime createdAt;
    private List<Assignment> assignments;
}
